unsafe/src/test/java/org/apache/spark/unsafe/types/UTF8StringSuite.java

```diff
@@ -392,6 +394,26 @@ public void levenshteinDistance() {
   }
 
   @Test
+  public void translate() {
+    assertEquals(fromString("1a2s3ae"),
+      fromString("translate").translate(ImmutableMap.of(
+        'r', '1',
+        'n', '2',
+        'l', '3',
+        't', '0'
+      )));
+    assertEquals(fromString("translate"),
+      fromString("translate").translate(new HashMap<Character, Character>()));
+    assertEquals(fromString("asae"),
+      fromString("translate").translate(ImmutableMap.of(
+        'r', '0',
+        'n', '0',
+        'l', '0',
+        't', '0'
+      )));
```

Comment:
We should put all the corner cases here, for example, Chinese, empty dict, or empty string.